src/components/DiveLog/DiveLog.js

```diff
@@ -2,6 +2,7 @@ import styled from 'styled-components';
 import { useState } from 'react';
 import { FaTrash } from 'react-icons/fa';
 import ScreenReaderOnly from '../ScreenReaderOnly';
+import DeleteModal from '../DeleteModal/DeleteModal';
 
 export default function DiveLog({
```

Comment:
There is a lot of repetition within your component. This first part is the same for both renders:
```js
       <DiveLogDetailsCard onClick={handleCardToggle}>
          <DiveNumberDetails>{divenumber}</DiveNumberDetails>
          <DetailsCardHead>
            <div>
              <p>{city}</p>
              <p> {country}</p>
              <p>{location}</p>
            </div>
            <BoxDateDelete>
              <Date>{date}</Date>
              <ButtonDelete>
                <FaTrash onClick={() => onDelete(_id)} />
                <FaTrash onClick={clickDelete} />
                <ScreenReaderOnly>delete</ScreenReaderOnly>
              </ButtonDelete>
            </BoxDateDelete>
          </DetailsCardHead>
```
I would recommend to only render the details if the `active` state is true. You then write the JSX for your card just once and don't repeat yourself.